plugin.program.AML/resources/utils_kodi_standalone.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+#
+# Advanced Emulator Launcher miscellaneous functions
+#
+
+# Copyright (c) 2016-2018 Wintermute0110 <wintermute0110@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+#
+# Replace functionality that depends on Kodi modules, so we can execute code
+# outside Kodi in standard Python.
+#
+
```

Comment:
as long as this is not used in kodi it is fine

Reply:
Some code of the Advanced Launchers needs to be run outside Kodi Python runtime. For example, to develop the databases, the scrapers, etc. using Kodi is tedious. Using CPython an a debugger is much faster. However, you gave me an idea to refurbish that file in a more elegant way.